### PR TITLE
Support changes in flow parsing

### PR DESCRIPTION
**Pull request by hzoo**
Just makes the tests pass (didn't spend that much time on this)

Ref original PR: https://github.com/babel/babel/pull/3385/files
Ref new babylon PR that was merged in 6.8.1 https://github.com/babel/babylon/pull/25/files

- [x] support new syntax
- [x] backwards compat?

### CODE REVIEW

File: packages/babel-generator/src/generators/types.js [hunk 1]
```diff
@@ -4,16 +4,6 @@
...
 export function Identifier(node: Object) {
-  // FIXME: We hang variance off Identifer to support Flow's def-site variance.
-  // This is a terrible hack, but changing type annotations to use a new,
-  // dedicated node would be a breaking change. This should be cleaned up in
-  // the next major.
-  if (node.variance === "plus") {
-    this.push("+");
-  } else if (node.variance === "minus") {
-    this.push("-");
-  }
```
Comment (loganfsmyth) [inline packages/babel-generator/src/generators/types.js]: Feels like we need to have this for backward-compatibility. Should we do

    if (node.variance) return TypeParameter(node);

Comment (hzoo) [inline packages/babel-generator/src/generators/types.js]: great idea, makes sense

**Comment by hzoo** [comment]
cc @samwgoldman @gabelevi

**Comment by hzoo** [comment]
This is blocking other PRs due to failing tests in babel on master due so we should do a patch release

@gabelevi, @glenjamin just b/c

**Pull request merged**